site/gatsby-site/src/components/RelatedIncidents.js

```diff
@@ -45,14 +45,20 @@ const reportsWithIncidentIds = async (reports, client) => {
     },
   });
 
-  return reports.map((report) => ({
-    ...report,
-    incident_id: response.data.incidents.filter((incident) =>
+  return reports.map((report) => {
+    const incident = response.data.incidents.find((incident) =>
       incident.reports
         .map((incidentReport) => incidentReport.report_number)
         .includes(report.report_number)
-    )[0]?.incident_id,
-  }));
+    );
+
+    return {
+      ...report,
+      incident_id: incident?.incident_id,
+      title: report.title,
```

Comment:
nit: place the `incident_*` fields next to each other (last?) for readability and grouping optional fields.